### ISSUE TITLE
Avoid throwing exceptions on span interactions

### ISSUE THREAD

**Issue by BhuvanB** [lightstep/lightstep-tracer-csharp#76]
While trying to tag or log on a finished span is invalid, there are ways where that may end up happening. While logging a warning somewhere might be warranted, an exception being thrown causes more damage than help.

I recently came across this exception working with OpenTracing.Contrib.NetCore:

```
(Inner Exception #0) System.InvalidOperationException: Adding logs System.Collections.Generic.Dictionary`2[System.String,System.Object] at 9/26/2019 1:41:19 PM -04:00 to already finished span.
   at LightStep.Span.CheckIfSpanFinished(String format, Object[] args) in /root/project/src/LightStep/Span.cs:line 236
   at LightStep.Span.Log(DateTimeOffset timestamp, IEnumerable`1 fields) in /root/project/src/LightStep/Span.cs:line 145
   at LightStep.Span.Log(IEnumerable`1 fields) in /root/project/src/LightStep/Span.cs:line 136
   at OpenTracing.Contrib.NetCore.Logging.OpenTracingLogger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter) in C:\projects\csharp-netcore\src\OpenTracing.Contrib.NetCore\Logging\OpenTracingLogger.cs:line 132
   at Microsoft.Extensions.Logging.Logger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
```

While it seems there might be a bug or incompatibility between OpenTracing.Contrib.NetCore and version of EF Core being used, I'm hoping we can eliminate (or at least provide option to do so) possibility of exceptions being thrown by Span operations and treat them as NoOp

Because of the dynamic way we instrument at times, I am hoping, we can extend this to the case where empty string or null is passed as tag key.

This was noticed on ```.NET Core 2.2``` and with ```LightStep 0.4.5```, though this is an issue likely in every version of LightStep on any platform.

**Comment by BhuvanB** [comment]
I have made a recommended change in my Pull Request: https://github.com/lightstep/lightstep-tracer-csharp/pull/77

I've limited it to just not throw exceptions on finished spans and didn't pollute the change with my recommendation of empty key call to SetTag

**Comment by austinlparker** [comment]
Released in [0.5.0](https://www.nuget.org/packages/LightStep/0.5.0) -- thanks for the contribution!
